http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/api/TestLeoRoutes.scala

```diff
@@ -122,14 +137,48 @@ trait TestLeoRoutes {
   val statusService =
     new StatusService(mockGoogleDataprocDAO, mockSamDAO, dbRef, applicationConfig, pollInterval = 1.second)
   val timedUserInfo = defaultUserInfo.copy(tokenExpiresIn = tokenAge)
-  val leoRoutes = new LeoRoutes(leonardoService, proxyService, statusService, swaggerConfig, contentSecurityPolicy)
-  with MockUserInfoDirectives {
+  val corsSupport = new CorsSupport(contentSecurityPolicy)
+  val proxyRoutes = new ProxyRoutes(proxyService, corsSupport)
+  val statusRoutes = new StatusRoutes(statusService)
+  val userInfoDirectives = new MockUserInfoDirectives {
     override val userInfo: UserInfo = defaultUserInfo
   }
-  val timedLeoRoutes = new LeoRoutes(leonardoService, proxyService, statusService, swaggerConfig, contentSecurityPolicy)
-  with MockUserInfoDirectives {
+  val leoRoutes = new LeoRoutes(leonardoService, userInfoDirectives)
+  val timedUserInfoDirectives = new MockUserInfoDirectives {
     override val userInfo: UserInfo = timedUserInfo
   }
+  val timedLeoRoutes = new LeoRoutes(leonardoService, timedUserInfoDirectives)
+
+  val runtimeService = new RuntimeServiceInterp(
+    blocker,
+    semaphore,
+    RuntimeServiceConfig(Config.proxyConfig.proxyUrlBase,
+                         imageConfig,
+                         autoFreezeConfig,
+                         dataprocConfig,
+                         Config.gceConfig),
+    whitelistAuthProvider,
+    serviceAccountProvider,
+    new MockDockerDAO,
+    FakeGoogleStorageInterpreter,
+    QueueFactory.makePublisherQueue()
+  )
+  val httpRoutes = new HttpRoutes(
+    swaggerConfig,
+    statusService,
+    proxyService,
+    leonardoService,
+    runtimeService,
+    userInfoDirectives,
+    contentSecurityPolicy
+  )
+  val timedHttpRoutes = new HttpRoutes(swaggerConfig,
+                                       statusService,
+                                       proxyService,
+                                       leonardoService,
+                                       runtimeService,
+                                       timedUserInfoDirectives,
+                                       contentSecurityPolicy)
```

Comment:
Took me some staring to notice the difference between `httpRoutes` and `timedHttpRoutes`. Would have been easier with 

```scala
val timedHttpRoutes = httpRoutes.copy(userInfoDirectives = timedUserInfoDirectives)
```

with `HttpRoutes` turned into a `case class`.

Reply:
case class are usually used for data types....`HttpRoutes` is not a data type at all